pkg/reconciler/revision/revision.go

```diff
@@ -144,17 +144,27 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1alpha1.Revision
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1alpha1.ContainerMissing,
-			v1alpha1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
-	}
 
-	rev.Status.ImageDigest = digest
+	if rev.Status.ImageDigests == nil {
+		rev.Status.ImageDigests = make(map[string]string)
+	}
+	for i := range rev.Spec.Containers {
```

Comment:
```suggestion
	for _, container := range rev.Spec.Containers {
```

and then use `container.Image`, `container.Ports` etc. below. Should be the same but a lot shorter.